gobblin-utility/src/main/java/gobblin/util/binpacking/WorstFitDecreasingBinPacking.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.binpacking;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.base.Predicate;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.MinMaxPriorityQueue;
+
+import gobblin.source.workunit.MultiWorkUnit;
+import gobblin.source.workunit.WorkUnit;
+
+import javax.annotation.Nullable;
+
+
+/**
+ * Implements a bin packing algorithm similar to worst-fit decreasing bin packing. Given a input list of {@link WorkUnit}s,
+ * a {@link Weighter} function, and a maximum weight per {@link MultiWorkUnit}, packs the input work units into
+ * {@link MultiWorkUnit}s.
+ *
+ * <p>
+ *   The algorithm is as follows:
+ *   * Sort work units decreasing by weight.
+ *   * Compute the minimum number of {@link MultiWorkUnit}s needed, and create them.
+ *   * For each work unit, find the {@link MultiWorkUnit} with the largest space available, if it fits, add it there,
+ *     otherwise, create a new {@link MultiWorkUnit} and add the work unit there.
+ * </p>
+ */
+public class WorstFitDecreasingBinPacking {
+
+  public static final String TOTAL_MULTI_WORK_UNIT_WEIGHT = "binpacking.multiWorkUnit.totalWeight";
+
+  public List<WorkUnit> pack(List<WorkUnit> workUnitsIn, Weighter weighter, long maxWeigthPerUnit) {
```

Comment:
`WorstFitDecreasingBinPacking` should implement `WorkUnitBinPacking` interface. The interface can have a `public List<WorkUnit> pack(List<WorkUnit> workUnitsIn, Weighter weighter, long maxWeigthPerUnit)` method